pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -239,12 +239,20 @@ func (r *ReconcilePravegaCluster) deploySegmentStore(p *pravegav1alpha1.PravegaC
 		return err
 	}
 
+	cm := &corev1.ConfigMap{}
+	name := p.Spec.Pravega.SegmentStoreEnvVars
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, cm)
 	configMap := pravega.MakeSegmentstoreConfigMap(p)
 	controllerutil.SetControllerReference(p, configMap, r.scheme)
 	err = r.client.Create(context.TODO(), configMap)
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
+	flag := r.compareKeys(cm, configMap)
+	if !flag {
+		log.Println("Rejecting the environment variables provided")
+	}
+	util.SetConfigMapNameForSSEnvVars(name, flag)
```

Comment:
We should **not** be updating any global variables. You could create a new field under PravegaCluster Status to indicate if ConfigMap Validation passed or failed and use that to decide if the CM should be added to the pod or not. The correct way to do the validation is actually to use a validationWebHook, but that may take longer to implement, so for now we should be good with this...